projects/core/src/event/event.service.ts

```diff
@@ -173,13 +173,13 @@ export class EventService {
    *
    * Should be used only in dev mode.
    */
-  private validateEventStream<T>(
+  private getValidatedEventStream<T>(
     source$: Observable<T>,
     eventType: Type<T>
   ): Observable<T> {
     return source$.pipe(
       tap(event => {
-        if (!(event instanceof eventType)) {
+        if (event?.constructor !== eventType) {
```

Comment:
Is it intended?

Reply:
Yup. It's to match exact class. Not all the ancestor classes (which is the case for `instanceof`).

Reply:
Reverted to instanceof